components/eventing-controller/pkg/backend/eventmesh/eventmesh.go

```diff
@@ -0,0 +1,364 @@
+	EventMeshTypeSegmentsLimit = 7
+	subscriptionNameLogKey     = "eventMeshSubscriptionName"
+	errorLogKey                = "error"
+)
+
+// Perform a compile time check.
+var _ Backend = &EventMesh{}
+
+type Backend interface {
+	// Initialize should initialize the communication layer with the messaging backend system
+	Initialize(cfg env.Config) error
+
+	// SyncSubscription should synchronize the Kyma eventing subscription with the subscriber infrastructure of messaging backend system.
+	// It should return true if Kyma eventing subscription status was changed during this synchronization process.
+	SyncSubscription(subscription *eventingv1alpha2.Subscription, cleaner cleaner.Cleaner, apiRule *apigatewayv1beta1.APIRule) (bool, error)
+
+	// DeleteSubscription should delete the corresponding subscriber data of messaging backend
+	DeleteSubscription(subscription *eventingv1alpha2.Subscription) error
+}
+
+func NewEventMesh(credentials *backendbebv1.OAuth2ClientCredentials, mapper backendutils.NameMapper, logger *logger.Logger) *EventMesh {
+	return &EventMesh{
+		oAuth2credentials: credentials,
+		logger:            logger,
+		SubNameMapper:     mapper,
+	}
+}
+
+type EventMesh struct {
+	client            client.PublisherManager
+	webhookAuth       *types.WebhookAuth
+	protocolSettings  *eventingv1alpha2.ProtocolSettings
+	namespace         string
+	eventMeshPrefix   string
+	oAuth2credentials *backendbebv1.OAuth2ClientCredentials
+	SubNameMapper     backendutils.NameMapper
+	logger            *logger.Logger
+}
+
+func (em *EventMesh) Initialize(cfg env.Config) error {
+	if em.client == nil {
+		authenticatedClient := auth.NewAuthenticatedClient(cfg)
+		httpClient, err := httpclient.NewHTTPClient(cfg.BEBAPIURL, authenticatedClient)
+		if err != nil {
+			return err
+		}
+		em.client = client.NewClient(httpClient)
+		em.webhookAuth = getWebHookAuth(cfg, em.oAuth2credentials)
+		em.protocolSettings = &eventingv1alpha2.ProtocolSettings{
+			ContentMode:     &cfg.ContentMode,
+			ExemptHandshake: &cfg.ExemptHandshake,
+			Qos:             &cfg.Qos,
+		}
+		em.namespace = cfg.BEBNamespace
+		em.eventMeshPrefix = cfg.EventTypePrefix
+	}
+	return nil
+}
+
+// getWebHookAuth returns the webhook auth config from the given env config
+// or returns an error if the env config contains invalid grant type or auth type.
+func getWebHookAuth(cfg env.Config, credentials *backendbebv1.OAuth2ClientCredentials) *types.WebhookAuth {
+	return &types.WebhookAuth{
+		ClientID:     credentials.ClientID,
+		ClientSecret: credentials.ClientSecret,
+		TokenURL:     cfg.WebhookTokenEndpoint,
+		Type:         types.AuthTypeClientCredentials,
+		GrantType:    types.GrantTypeClientCredentials,
+	}
+}
+
+// SyncSubscription synchronize the EV2 subscription with the EMS subscription.
+// It returns true, if the EV2 subscription status was changed.
+func (em *EventMesh) SyncSubscription(subscription *eventingv1alpha2.Subscription, cleaner cleaner.Cleaner,
```

Comment:
The function looks really good now! Good job and also thanks for writing tests for it!